crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -946,18 +947,16 @@ impl<DB: Database, EF: ExecutorFactory> BlockchainTree<DB, EF> {
         let canonical = self.split_chain(chain_id, chain, ChainSplitTarget::Hash(*block_hash));
         durations_recorder.record_relative(MakeCanonicalAction::SplitChain);
 
-        let mut block_fork = canonical.fork_block();
-        let mut block_fork_number = canonical.fork_block_number();
+        let mut fork_block = canonical.fork_block();
         let mut chains_to_promote = vec![canonical];
 
         // loop while fork blocks are found in Tree.
-        while let Some(chain_id) = self.block_indices().get_blocks_chain_id(&block_fork.hash) {
-            let chain = self.state.chains.remove(&chain_id).expect("To fork to be present");
-            block_fork = chain.fork_block();
+        while let Some(chain_id) = self.block_indices().get_blocks_chain_id(&fork_block.hash) {
+            let chain = self.state.chains.remove(&chain_id).expect("fork is present");
             // canonical chain is lower part of the chain.
             let canonical =
-                self.split_chain(chain_id, chain, ChainSplitTarget::Number(block_fork_number));
-            block_fork_number = canonical.fork_block_number();
+                self.split_chain(chain_id, chain, ChainSplitTarget::Number(fork_block.number));
+            fork_block = canonical.fork_block();
```

Comment:
this change preserves old behavior because `chain.fork_block()` equals to `canonical.fork_block()` as `canonical` is just a lower part of the `chain`